### ISSUE TITLE
WebAssembly Encoding.UTF8.GetString extremely slow on cyrillic characters

### ISSUE THREAD

**Issue by Zonnery** [dotnet/runtime#54493]
### Description

**Encoding.UTF8.GetString** extremely slow on WebAssembly than on serverside NET Core 5 console app if string contains any cyrillic symbols.
As this method uses for calling **HttpClient.GetStringAsync()**, **WebAssembly** App have serious performance issues and UI thread hangup.

Steps to reproduce:
Create any blazor project from template and put code below in **OnInitializedAsync**

```
		var data = Encoding.UTF8.GetBytes(string.Join("",Enumerable.Repeat("Ы",1000000)));
		Stopwatch stopwatch = new Stopwatch();
		stopwatch.Start();
		Encoding.UTF8.GetString(data);
		stopwatch.Stop();
```

Stopwatch will be equals around **1100ms** (for me).

Then change it to 

 ```
		var data = Encoding.UTF8.GetBytes(string.Join("",Enumerable.Repeat("S",1000000)));
		Stopwatch stopwatch = new Stopwatch();
		stopwatch.Start();
		Encoding.UTF8.GetString(data);
		stopwatch.Stop();
```

Stopwatch will be equals around **68ms** (for me).

On **NET Core 5** console app both case approximitly equals.

### Configuration

.NET SDK (reflecting any global.json):
 Version:   5.0.301
 Commit:    ef17233f86

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.301\

Host (useful for support):
  Version: 5.0.7
  Commit:  556582d964

**Comment by lewing** [comment]
There have been several improvements in this area since net5 @radekdoulik can you do some comparisons on main and past e the results?

**Comment by lewing** [comment]
@radekdoulik please update the status here

**Comment by lewing** [comment]
cc @kg

**Comment by kg** [comment]
Based on the date of this issue, I assume it was not using AOT. So AOT should offer big improvements here. I would also expect jiterp to help. I'll see whether we already have a benchmark for this in dotnet/performance.

**Comment by kg** [comment]
One of the automated benchmarks does UTF8.GetString for a big blob of cyrillic text, so I believe this is covered. However, I don't see it in the automated reports - the most similar results link https://pvscmdupload.blob.core.windows.net/reports/allTestHistory/refs/heads/main_x64_ubuntu%2018.04_CompilationMode=wasm_RunKind=micro/System.Text.Tests.Perf_Encoding.GetBytes(size:%20512,%20encName:%20 is broken and I can't see one that operates on the big cyrillic test file. Do you know who I would ping to follow up on this?

**Comment by kg** [comment]
A manual run of the Utf8 benchmarks generates timings that look about right in interpreter mode, and Jiterpreter statistics appear to indicate that we JIT this code fine. So I think we can close this unless the reporter is still having issues, especially since AOT should work well for this scenario.

|       Method |              Input |       Mean |    Error |   StdDev |     Median |
|------------- |------------------- |-----------:|---------:|---------:|-----------:|
| GetByteCount |    EnglishAllAscii | 1,224.6 us |  9.89 us |  9.25 us | 1,222.4 us |
|     GetBytes |    EnglishAllAscii | 1,498.7 us | 28.65 us | 26.80 us | 1,489.5 us |
|    GetString |    EnglishAllAscii |   975.6 us | 18.29 us | 16.22 us |   971.8 us |
| GetByteCount | EnglishMostlyAscii | 4,386.1 us | 85.28 us | 79.77 us | 4,390.4 us |
|     GetBytes | EnglishMostlyAscii | 4,399.3 us | 28.38 us | 23.70 us | 4,406.0 us |
|    GetString | EnglishMostlyAscii | 1,179.3 us | 11.47 us | 10.16 us | 1,179.1 us |
| GetByteCount |            Chinese | 2,107.3 us |  5.70 us |  5.33 us | 2,107.6 us |
|     GetBytes |            Chinese | 2,536.3 us |  7.31 us |  6.11 us | 2,537.8 us |
|    GetString |            Chinese | 1,658.1 us |  4.95 us |  4.39 us | 1,658.1 us |
| GetByteCount |           Cyrillic | 1,840.7 us | 10.76 us | 10.07 us | 1,842.8 us |
|     GetBytes |           Cyrillic | 2,143.5 us |  7.32 us |  6.85 us | 2,143.2 us |
|    GetString |           Cyrillic |   431.8 us | 15.22 us | 17.52 us |   422.1 us |
| GetByteCount |              Greek | 2,329.2 us |  6.69 us |  5.93 us | 2,329.2 us |
|     GetBytes |              Greek | 2,814.0 us | 12.82 us | 11.37 us | 2,812.3 us |
|    GetString |              Greek |   736.5 us | 26.77 us | 30.83 us |   743.3 us |

**Comment by lewing** [comment]
Closing, @Zonnery feel free to reopen if you feel this hasn't been adequately addressed in dotnet 8.0 preview 6+
